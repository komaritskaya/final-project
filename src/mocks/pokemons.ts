import { Pokemon } from '../types';

const pokemons: Pokemon[] = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: true,
    catchDate: new Date(),
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
];

export default pokemons;
