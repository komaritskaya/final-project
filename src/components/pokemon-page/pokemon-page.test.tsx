import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import PokemonPage from './pokemon-page';
import pokemons from '../../mocks/pokemons';

const mockStore = configureStore([]);

it('Should PokemonPage component render correctly', () => {
  const store = mockStore({
    data: {
      pokemons,
    },
  });

  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <PokemonPage />
        </Router>
      </Provider>,
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
