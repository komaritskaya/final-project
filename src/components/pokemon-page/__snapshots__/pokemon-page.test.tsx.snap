// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PokemonPage component render correctly 1`] = `
<div
  className="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthXl"
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "inherit",
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2 MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        Back
      </span>
    </button>
  </a>
  <div
    className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
  >
    <img
      alt="bulbasaur (id: 1)"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      src="/img/1.png"
      title="bulbasaur (id: 1)"
    />
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        BULBASAUR (ID: 1)
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Caught on 11/25/2020
      </p>
    </div>
  </div>
</div>
`;
